### PR TITLE
 # 169817552 add user role settings

### PR DESCRIPTION
**Pull request by higustave-ops**
#### What does this PR do?
`create user role settings endpoint`
#### Description of Task to be completed?
`To create user role settings endpoint feature by allowing a super admin to be able to update/reset other user's roles`
#### How should this be manually tested?
* Clone this repo
* Navigate in the repo root using `CMD` or any `TERMINAL`
* Install all packages using `npm install`
* Type `npm run dev` for Postman testing And `npm run test` for mocha testing
* Open postman choose PATCH `localhost:3000/api/admin/reset-role`
* Click `body` then `x-www-form-urlencoded`
* Fill in `userEmail` and `newRole`
* Click Headers and Type `Authorization` paste in your `After logging in Token` as value
* Click `SEND` on the top right corner of Postman

#### What are the relevant pivotal tracker stories?
- [#169817552](https://www.pivotaltracker.com/story/show/169817552)

#### Screenshots (if appropriate)
`Postman screenshot`
Screen Shot 2020-01-08 at 15 50 44

### CODE REVIEW

File: src/controllers/settings.controller.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  * @param {object} res response object
+  * @memberof UserRoleSettingsController
+  * @returns {object} ResponseService
+  */
+  static async changeUserRole(req, res) {
+    const { userEmail, userRole } = req.body;
+    const findUser = await UserService.findUserByProperty({ email: userEmail.trim() });
+    if (findUser && findUser.role !== 'super_admin') {
+      await UserService.updateUser({ email: userEmail.trim() }, { role: userRole.trim() });
+      const data = await UserService.findUserByProperty({ email: userEmail.trim() });
```
Comment (adesege) [inline src/controllers/settings.controller.js]: You don't have to do another find here. You can return the `findUser` on line 20 and override the role property with the new userRole that was updated.

File: src/routes/settings.route.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import authMiddleware from '../middlewares/auth.middleware';
+import SettingsController from '../controllers/settings.controller';
+import { validateUserRole } from '../validations/role.validation';
+
+const router = express.Router();
+
+router.patch('/reset-roles', authMiddleware.checkUserLoggedIn, validateUserRole, authMiddleware.verifyIfUserIsAdmin, SettingsController.changeUserRole);
```
Comment (adesege) [inline src/routes/settings.route.js]: You can update this to `/roles` instead.

File: src/controllers/settings.controller.js (hunk 1 above)
Comment (higustave-ops) [reply src/controllers/settings.controller.js]: finished now

File: src/routes/settings.route.js (hunk 2 above)
Comment (higustave-ops) [reply src/routes/settings.route.js]: finished now

File: src/controllers/settings.controller.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  * @memberof UserRoleSettingsController
+  * @returns {object} ResponseService
+  */
+  static async changeUserRole(req, res) {
+    const { userEmail, userRole } = req.body;
+    const findUser = await UserService.findUserByProperty({ email: userEmail.trim() });
+    if (findUser && findUser.role !== 'super_admin') {
+      await UserService.updateUser({ email: userEmail.trim() }, { role: userRole.trim() });
+      const data = findUser;
+      data.role = userRole;
```
Comment (adesege) [inline src/controllers/settings.controller.js]: These lines are unnecessary since you are only returning a role property in your response.

**Review by adesege — APPROVED** [review]

**Pull request merged**